notes/zrq/20200703-01-project-plan.txt

```diff
@@ -0,0 +1,308 @@
+                    - Login to Zeppelin any time during that period.
+                    - Resources allocated based on the booking.
+
+                - Job queue
+
+                    - Queue for Spark jobs, inside the notebook.
+                        - Custom queue UI on top of the standard YARN interface ?
+
+                - Access queue
+                    - Queue for Zeppelin access.
+
+                        - First come first served.
+                        - No time limit, use the system until done.
+                        - Queue for access when busy.
+                        - Notification system when you reach the head of the queue.
+
+                - Resource sharing
+
+                    - Shared Zeppelin environment.
+                        - Zeppelin accounts for isolation.
+                        - How to manage isolation.
+                            - Separate Unix accounts ?
+
+                    - Shared Spark environment.
+                        - How to manage isolation.
+                            - Separate Unix accounts ?
+
+                    - Seperate Zeppelin Pod.
+                        - Shared Kubernetes cluster.
+                        - Shared Openstack VMs
+                        - High load from one will take resources from others
+                        - OutOfMemory, disc or CPU still a possibility.
+                        - Account isolation handled by the Pod boundaries.
+                            - No need for synchronised Unix accounts.
+                            - Pod limits on cpu and memory.
+
+                    - Separate Spark Pod.
+                        - Shared Kubernetes cluster.
+                        - Shared Openstack VMs
+                        - High load from one will take resources from others
+                        - OutOfMemory, disc or CPU still a possibility.
+                        - Account isolation handled by the Pod boundaries.
+                            - No need for synchronised Unix accounts.
+                            - Pod limits on cpu and memory.
+
+                    - Separate Kubernetes cluster.
+                        - Power user, requires booking.
+                        - Workshop group, requires booking.
+                        - Available to use now if there is space.
+                        - 10 min to create the cluster
+
+                        - Separate Openstack VMs
+                        - Account isolation handled by the Pod boundaries.
+                            - No need for synchronised Unix accounts.
+                            - Pod limits on cpu and memory.
+
+
+                    * I think we need parts of all of these.
+
+                    * MVP use cases need more detail.
+                        - What the queue is for and how it is implemented ?
+
+                    * Comments are more detailed than the wiki page they are commenting on :-(
```

Comment:
Is this still an issue? Do we need more details on the wiki?

Reply:
The calendar system isn't a definite yet, we need to get a prototype working. The queues idea has a lot of merit and we may end up combining aspects of both.

Discussions during our telecons suggest there probably are two queues:
- A queue for launching Spark jobs
- A queue for launching a Zeppelin instance

I think we still have some questions about these:
- What effect does the Spark queue have on interactive response of the notebooks ?
- How does the user interact with the Zeppelin queue ?

Reply:
If the queues are fast, then the user cam wait  ... but if the queues are long, then do we need a notification system to tell the users that their resources are ready ?